src/Symfony/Component/Ldap/Security/CheckLdapCredentialsListener.php

```diff
@@ -72,6 +69,9 @@ public function onCheckPassport(CheckPassportEvent $event)
         }
 
         $user = $passport->getUser();
+        if (!$user instanceof LdapUser) {
+            return;
```

Comment:
> The CheckLdapCredentialsListener does nothing when the user is not an instance of LdapUser.

This is major BC in the logic. The user can be loaded from the database, but at the same time the password is checked in LDAP.

in my opinion, it's more logical to check LDAP in the first, and if the check is failed fallback to default credentials checker.
For example my [CheckLdapCredentialsListener](https://github.com/vtsykun/packeton/blob/2.3/src/Security/CheckLdapCredentialsListener.php) implementation.

Reply:
I don't know if the  BC is "major" or not, but checking LDAP first and fallback on another credential system increases the attack surface IMHO. But you're right about the BC, I added a BC layer.